pkg/reconciler/tenancy/workspace/workspace_reconcile_scheduling.go

```diff
@@ -150,19 +147,13 @@ func (r *schedulingReconciler) reconcile(ctx context.Context, workspace *tenancy
 			return reconcileStatusStopAndRequeue, err
 		}
 
-		// now complete the second part of our two-phase commit: set location in workspace
-		u, err := url.Parse(shard.Spec.ExternalURL)
-		if err != nil {
-			// shouldn't happen since we just checked in isValidShard
-			conditions.MarkFalse(workspace, tenancyv1alpha1.WorkspaceScheduled, tenancyv1alpha1.WorkspaceReasonReasonUnknown, conditionsv1alpha1.ConditionSeverityError, "Invalid connection information on target Shard: %v.", err)
-			return reconcileStatusStopAndRequeue, err // requeue
-		}
-
-		u.Path = path.Join(u.Path, clusterName.Path().RequestPath())
-		workspace.Status.URL = u.String()
+		// We've done the two phase commit, set the workspace as scheduled.
 		workspace.Status.Cluster = clusterName.String()
 		conditions.MarkTrue(workspace, tenancyv1alpha1.WorkspaceScheduled)
```

Comment:
I think this PR is the wrong direction. We should rather set cluster and URL (in spec) here (that's what scheduling is), and then set the condition in another pass. And this other pass can be in this very same reconciler.

Reply:
Yeah that's what we had before, see https://github.com/kcp-dev/kcp/pull/2557#discussion_r1063510107 for reasoning on splitting it up